### ISSUE TITLE
When using the chat-messages API, the “inputs” value seems to be cached and cannot be overwritten if the conversation is ongoing.

### ISSUE THREAD

**Issue by mihit** [langgenius/dify#11140]
### Self Checks

- [X] This is only for bug report, if you would like to ask a question, please head to [Discussions](https://github.com/langgenius/dify/discussions/categories/general).
- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] [FOR CHINESE USERS] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### Dify version

0.11.2

### Cloud or Self Hosted

Self Hosted (Docker)

### Steps to reproduce

Just do as the title.

### ✔️ Expected Behavior

I want to overwrite inputs.

### ❌ Actual Behavior

Never.

**Comment by crazywoola** [comment]
Duplicated

**Comment by liuzixiao666** [comment]
> Duplicated, see the bot's reply, we are still reviewing this feature.

Can you reveal why this feature has been under discussion? What are the negative effects of this feature? I think it is a very important feature, and related competitors such as Coze, FastGPT, Maxkb all have related functions
My understanding here is that 'query' serves as the actual input of the user, while 'input' should be my explanation of the user's state. This is beneficial for the workflow to understand the current process that should be running and ensure that the user's actual dialogue input is not contaminated
For example, 'query'=[happy] hello, the workflow may reply with 'care' hello!, But I don't want [happiness] to pollute the workflow's replies (mainly due to memory window issues)

**Comment by mihit** [comment]
@liuzixiao666 
I would like to discuss the following points:

- **APIs should clearly define their specifications** as a fundamental principle.
- The name `inputs` does not explicitly indicate that input values are being cached. If this is the intended behavior, a more appropriate name such as `persist_inputs` would be better.
- Even though `inputs` are being cached and new input values are being discarded, the system does not inform the user about this.
In other words, while I understand that caching inputs can be beneficial for certain use cases, I believe there are issues with the current implementation from a software perspective.

**Comment by liuzixiao666** [comment]
> @liuzixiao666 I would like to discuss the following points:
> 
> * **APIs should clearly define their specifications** as a fundamental principle.
> * The name `inputs` does not explicitly indicate that input values are being cached. If this is the intended behavior, a more appropriate name such as `persist_inputs` would be better.
> * Even though `inputs` are being cached and new input values are being discarded, the system does not inform the user about this.
>   In other words, while I understand that caching inputs can be beneficial for certain use cases, I believe there are issues with the current implementation from a software perspective.

thanks！！！！！

**Comment by glacierck** [comment]
It should not be cached. This is the only way to control the multi branch behavior flow by passing parameters from the outside. If you cache the outside, you cannot intervene in the session flow. And why cache it? Is there any problem with not caching?

**Comment by glacierck** [comment]
My RAG has multiple knowledge base branches that support users to select the loaded knowledge base from external settings and switch within the same Q&A process. Cache operations make everything impossible.

**Comment by alexbalandi** [comment]
Is there still no solution for this? :( It seems my only way to pass some state with message is to directly modify the query. Is there maybe some other way? This really hurts my use-case rn (integration with chatwoot where I need to do certain things depending on status of convo in chatwoot)

**Comment by liuzixiao666** [comment]
> Is there still no solution for this? :( It seems my only way to pass some state with message is to directly modify the query. Is there maybe some other way? This really hurts my use-case rn (integration with chatwoot where I need to do certain things depending on status of convo in chatwoot)

The latest version of dify has solved this problem
